packages/mobile/src/screens/edit-track-screen/fields/AccessAndSaleField/PremiumRadioField/TrackPriceField.tsx

```diff
@@ -0,0 +1,43 @@
+import { useField } from 'formik'
+
+import { Text } from 'app/components/core'
+
+import { BoxedTextField } from './BoxedTextField'
+
+const TRACK_PRICE = 'premium_conditions.usdc_purchase.price'
+
+const messages = {
+  title: 'Set a Price',
+  description:
+    'Set the price fans must pay to unlock this track (minimum price of $0.99)',
+  label: 'Cost to Unlock',
+  placeholder: '0.99',
+  dollars: '$',
+  usdc: '(USDC)'
+}
+
+export const TrackPriceField = () => {
+  const [{ value }] = useField(TRACK_PRICE)
```

Comment:
if we're passing the name in anyway, we could push this down into `BoxedTextField`

Reply:
so im String(value) here cause value is a number, but has to be a string for react-natives text-input :) going to add some more logic here for proper human readable too